android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -491,24 +491,56 @@ constructor(
     gatewayModeHeaderValue: String? = null,
     searchPath: String,
     patientRelatedResourceTypes: MutableList<ResourceType>,
+    currentPage: Int = 1,
   ) {
-    val resultBundle =
-      if (gatewayModeHeaderValue.isNullOrEmpty()) {
-        fhirResourceDataSource.getResource(searchPath)
-      } else
-        fhirResourceDataSource.getResourceWithGatewayModeHeader(
-          gatewayModeHeaderValue,
-          searchPath,
-        )
+    val pageSize = 200
+    var nextPage = currentPage
+    var totalItemsProcessed = 0
+
+    while (true) {
+      val resultBundle = fetchResourceBundle(gatewayModeHeaderValue, searchPath, nextPage)
+      val itemsCount = resultBundle.entry.size
+      totalItemsProcessed += itemsCount
+
+      processResultBundleEntries(resultBundle.entry, patientRelatedResourceTypes)
+
+      when {
+        itemsCount == pageSize -> {
```

Comment:
This condition seems to terminate the loop after the hardcoded pagesize of 200 is reached, would the solution still work if we had more than 200 pages?